### PR TITLE
[MM-10062] Revert help note when team creation is set to system admin only

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Revert help note when team creation is set to system admin only

#### Ticket Link
Jira ticket: [MM-10062]

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx [hunk 1]
```diff
@@ -88,6 +88,7 @@ export default class UsersAndTeamsSettings extends AdminSettings {
 
     getConfigFromState = (config) => {
         config.TeamSettings.EnableUserCreation = this.state.enableUserCreation;
+        config.TeamSettings.EnableTeamCreation = this.state.enableTeamCreation;
```
Comment (saturninoabril) [inline components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx]: Not sure if this is the correct way to do this now that the role based permission took effect already.

**Comment by jespino** [comment]
With the new permissions, we'll not have EnableTeamCreation, because it comes from the permissions schemas. I'm not sure if we can maintain that text in the code. We haven't an easy way to know if permissions are only for system administrators or not, and in Phase 2, It will be harder. I don't know if we have to review that from the UX perspective.

**Comment by saturninoabril** [comment]
I'll get back on this PR just to further test as I'm seeing weird `undefined` value of `enableTeamCreation` when a user is a system admin. I'm specifically looking at the returned data of `loadRolesIfNeeded` for system admin as compared to regular user.

**Comment by saturninoabril** [comment]
@jespino This is now ready for review.

**Pull request merged**